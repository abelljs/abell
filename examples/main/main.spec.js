const fs = require('fs');
const path = require('path');

const {
  preTestSetup,
  getSelector
} = require('../../tests/test-utils/helpers.js');

const examplePath = path.join(__dirname, 'dist');

describe('examples/main', () => {
  beforeAll(async () => {
    await preTestSetup('main');
  });

  describe('dist/index.html', () => {
    /** Test dist/index.html */

    let indexSelector;
    const testComponents = [
      {
        it: 'should render basic maths and Abell Variables',
        selector: '[data-test="basic-test"]'
      },
      {
        it: 'should render contentArray loop',
        selector: '[data-test="contentarray-container"]'
      }
    ];

    beforeAll(() => {
      indexSelector = getSelector(path.join(examplePath, 'index.html'));
    });

    for (const testComponent of testComponents) {
      it(testComponent.it, () => {
        expect(indexSelector(testComponent.selector).html()).toMatchSnapshot();
      });
    }
  });

  // // another-blog index
  describe('another-blog/*', () => {
    /**
     * Tests-
     * another-blog/index.html
     * & another-blog/example.html
     */

    let indexSelector;
    let exampleSelector;
    const testComponents = [
      {
        it: 'should render base info of blog',
        selector: '[data-test="basic-info"]'
      },
      {
        it: 'should render blog content',
        selector: '[data-test="blog-content"]'
      }
    ];

    beforeAll(() => {
      indexSelector = getSelector(
        path.join(examplePath, 'another-blog', 'index.html')
      );

      exampleSelector = getSelector(
        path.join(examplePath, 'another-blog', 'example.html')
      );
    });

    for (const testComponent of testComponents) {
      it(testComponent.it, () => {
        expect(indexSelector(testComponent.selector).html()).toMatchSnapshot();
      });
    }

    it('should render name of blog in example.html', () => {
      expect(exampleSelector('body').html().trim()).toBe('another-blog');
    });
  });

  describe('new-blog/', () => {
    /**
     * Tests-
     * new-blog/*
     */
    it('should have index.html and example.html in new-blog', () => {
      expect(
        fs.existsSync(path.join(examplePath, 'new-blog', 'index.html'))
      ).toBe(true);

      expect(
        fs.existsSync(path.join(examplePath, 'new-blog', 'example.html'))
      ).toBe(true);
    });

    describe('sub-blog/', () => {
      it('should render expected blog content in index.html', () => {
        expect(
          getSelector(
            path.join(examplePath, 'new-blog', 'sub-blog', 'index.html')
          )('[data-test="blog-content"]').html()
        ).toMatchSnapshot();
      });

      it('should render path to the blog in example.html', () => {
        expect(
          getSelector(
            path.join(examplePath, 'new-blog', 'sub-blog', 'example.html')
          )('body')
            .html()
            .trim()
        ).toBe('new-blog/sub-blog');
      });
    });
  });

  // deep
  describe('deep/', () => {
    /**
     * Tests-
     * deep/index.html
     * & deep/moredeep/index.html
     */
    let deepIndex;
    let moreDeepIndex;

    beforeAll(() => {
      deepIndex = getSelector(path.join(examplePath, 'deep', 'index.html'));
      moreDeepIndex = getSelector(
        path.join(examplePath, 'deep', 'moredeep', 'index.html')
      );
    });

    it('should render html body text of deep/index.html', () => {
      expect(deepIndex('body').html()).toBe('ok ..');
    });

    it('should render html of deep/moredeep/index.html', () => {
      expect(moreDeepIndex('body').html()).toBe('../..');
    });
  });
});
