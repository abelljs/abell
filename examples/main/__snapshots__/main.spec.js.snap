// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/main another-blog/* should render base info of blog 1`] = `null`;

exports[`examples/main another-blog/* should render blog content 1`] = `
"
      <h1 id=\\"another-blog\\">Another blog</h1>
<p>6</p>
<p>Amazing blog right</p>
    "
`;

exports[`examples/main dist/index.html should render basic maths and Abell Variables 1`] = `
"
      <div>JS Test 11</div>
      <div>Include from JSON Test hi I am from JSON</div>
      <div>globalMeta Test Abell standard example</div>
      <div>root test </div>
      <div>path test </div>
    "
`;

exports[`examples/main dist/index.html should render contentArray loop 1`] = `
"
      contentArray Test
      <div>
        <div>
                <span class=\\"data-title\\">sub-blog (new-blog/sub-blog)</span>
                <span class=\\"data-createdat\\">true</span>
                <span class=\\"data-modifiedat\\">true</span>
                <span class=\\"data-foo\\">undefined</span>
              </div>
            
              <div>
                <span class=\\"data-title\\">new-blog (new-blog)</span>
                <span class=\\"data-createdat\\">true</span>
                <span class=\\"data-modifiedat\\">true</span>
                <span class=\\"data-foo\\">undefined</span>
              </div>
            
              <div>
                <span class=\\"data-title\\">Another blog (another-blog)</span>
                <span class=\\"data-createdat\\">true</span>
                <span class=\\"data-modifiedat\\">true</span>
                <span class=\\"data-foo\\">bar</span>
              </div>
      </div>
    "
`;

exports[`examples/main new-blog/ sub-blog/ should render expected blog content in index.html 1`] = `
"
      <h1 id=\\"inside-the-new-blog\\">Inside the new blog</h1>
<p>../..</p>
<p><a href=\\"hello\\">sub-blog/hello</a></p>
<p>{{ escape this blog }}</p>
<p>4</p>
    "
`;
